src/http/representation/RepresentationMetadata.ts

```diff
@@ -304,6 +306,52 @@ export class RepresentationMetadata {
     return this;
   }
 
+  private setContentTypeParams(input: ContentType | string | undefined): void {
+    if (input === undefined) {
+      this.removeContentTypeParameters();
+      return;
+    }
+    if (typeof input === 'string') {
+      input = parseContentTypeWithParameters(input);
+    }
+
+    // Make sure complete Content-Type RDF structure is gone
+    this.removeContentTypeParameters();
+
+    Object.entries(input.parameters ?? []).forEach(([ paramKey, paramValue ], idx): void => {
+      const paramNode = DataFactory.blankNode(`parameter${idx + 1}`);
+      this.addQuad(this.id, SOLID_META.terms.ContentTypeParameter, paramNode);
+      this.addQuad(paramNode, RDFS.terms.label, paramKey);
+      this.addQuad(paramNode, SOLID_META.terms.value, paramValue);
+    });
+  }
+
+  /**
+   * Parse the internal RDF structure to retrieve the Record with ContentType Parameters.
+   * @returns An Record&lt;string,string&gt; object with Content-Type parameters and their values.
+   */
+  private getContentTypeParams(): Record<string, string> | undefined {
+    const params = this.getAll(SOLID_META.terms.ContentTypeParameter);
+    return params.length > 0 ?
+      params.reduce((acc, term): Record<string, string> => {
+        const key = this.store.getObjects(term, RDFS.terms.label, null)[0].value;
+        const { value } = this.store.getObjects(term, SOLID_META.terms.value, null)[0];
+        return { ...acc, [key]: value };
+      }, {}) :
+      undefined;
+  }
+
+  private removeContentTypeParameters(): void {
+    const params = this.store.getQuads(this.id, SOLID_META.terms.ContentTypeParameter, null, null);
+    params.forEach((quad): void => {
+      const labels = this.store.getQuads(quad.object, RDFS.terms.label, null, null);
+      const values = this.store.getQuads(quad.object, SOLID_META.terms.value, null, null);
+      this.store.removeQuads(labels);
+      this.store.removeQuads(values);
+    });
```

Comment:
```suggestion
    const params = this.store.getQuads(this.id, SOLID_META.terms.ContentTypeParameter, null, null);
    for (const quad of params) {
      await endOfStream(this.removeMatches(quad.object, RDF.terms.label));
      await endOfStream(this.removeMatches(quad.object, SOLID_META.terms.value));
    }
```
In general standard for loops are preferred. We should probably add a linter rule for that if one exists. I have also never used the `removeMatches` function (just discovered it exists) so I might have suggested something wrong.

Reply:
Using await, would make the method async, implying I have to make other methods calling it async. (Like setContentParams(...))

Reply:
The double `await` in a loop worries me; if we need to await, should be parallel.

Reply:
Ah right, making it async is annoying, nevermind then. Would be great if N3.js had a similar sync function though *hint hint* 😉

Reply:
Looks like the internal implementation of `removeMatches` essentially does what the code above has. So only optimization I see is

```suggestion
      this.store.removeQuads([...labels, ...values]);
```
but that's very minor. And a `for` loop indeed.

Reply:
That is exactly what I just did now